airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java

```diff
@@ -218,9 +214,61 @@ private static Container getMain(final String image,
     return containerBuilder.build();
   }
 
-  private static void copyFilesToKubeConfigVolume(final KubernetesClient client,
-                                                  final Pod podDefinition,
-                                                  final Map<String, String> files) {
+  public static List<ContainerPort> createContainerPortList(final Map<Integer, Integer> internalToExternalPorts) {
+    return internalToExternalPorts.keySet().stream()
+        .map(integer -> new ContainerPortBuilder()
+            .withContainerPort(integer)
+            .build())
+        .collect(Collectors.toList());
+  }
+
+  public static void copyFilesToKubeConfigVolumeMain(final Pod podDefinition, final Map<String, String> files) {
```

Comment:
if this is only used in the async pod process, should this block be moved to that class?